graph-proxy/src/graphql/workflows.rs

```diff
@@ -74,71 +63,65 @@ enum WorkflowStatus {
 }
 
 impl WorkflowStatus {
-    /// Creates a new `WorkflowStatus` from `IoArgoprojWorkflowV1alpha1WorkflowStatus` and associated metadata.
+    /// Creates a new [`WorkflowStatus`] from [`IoArgoprojWorkflowV1alpha1Workflow`]
     fn new(
-        value: IoArgoprojWorkflowV1alpha1WorkflowStatus,
-        metadata: Arc<Metadata>,
+        workflow: Arc<IoArgoprojWorkflowV1alpha1Workflow>,
     ) -> Result<Option<Self>, WorkflowParsingError> {
-        match value.phase.as_deref() {
-            Some("Pending") => Ok(Some(Self::Pending(WorkflowPendingStatus::from(value)))),
-            Some("Running") => Ok(Some(Self::Running(WorkflowRunningStatus::new(
-                value, metadata,
-            )?))),
+        let status = workflow.status.as_ref().unwrap();
```

Comment:
Instead of calling `unwrap` here could we return a variant of `WorkflowParsingError` as `unwrap` will cause the thread to panic?

Could we also pass the `status` to the `Workflow{Pending,Running,Complete}Status` constructors and avoid their `unwrap`s? You may need to `clone` the `status`, but it should be possible to pass the reference to it with an appropriate lifetime

Reply:
First one - done.

Second one - was a fun journey to do this; what I've done is likely not perfect, but it's in the ballpark of what was suggested I think.